boring/src/ssl/mod.rs

```diff
@@ -1702,19 +1740,32 @@ impl SslContextBuilder {
 
     /// Returns a shared reference to the context's certificate store.
     #[corresponds(SSL_CTX_get_cert_store)]
+    #[deprecated(note = "This reference is useless. Use build() instead.")]
     pub fn cert_store(&self) -> &X509StoreBuilderRef {
         #[cfg(feature = "rpk")]
         assert!(!self.is_rpk, "This API is not supported for RPK");
 
+        // It's not safe to return a shared &X509Store when the builder owns it exclusively
         unsafe { X509StoreBuilderRef::from_ptr(ffi::SSL_CTX_get_cert_store(self.as_ptr())) }
     }
 
     /// Returns a mutable reference to the context's certificate store.
+    ///
+    /// ## Panics
+    ///
+    /// * If the a shared store has been set via [`set_cert_store_ref`]
```

Comment:
Let's not make it panic and let's wait for boring 5 to introduce a breaking change here.